crates/debug/types/src/source_file.rs

```diff
@@ -625,18 +625,27 @@ impl SourceContent {
         line_index: LineIndex,
         column_index: ColumnIndex,
     ) -> Option<ByteIndex> {
-        let column_index = column_index.to_usize();
         let line_span = self.line_range(line_index)?;
         let line_src = self
             .content
             .get(line_span.start.to_usize()..line_span.end.to_usize())
             .expect("invalid line boundaries: invalid utf-8");
-        if line_src.len() < column_index {
+
+        let col_chars = column_index.to_usize();
+        let num_chars = line_src.chars().count();
+
+        if col_chars > num_chars {
```

Comment:
It isn't necessary to check this here, as  `.char_indices().nth()` will produce `None` if the index is out of bounds of the line.